docs/docs/60-new-docs/50-user-guide/60-reference-docs/50-analysis-templates.md

```diff
@@ -0,0 +1,216 @@
+
+The following example continually polls a Prometheus server to get the total number of errors (i.e., HTTP response code >= 500) every 5 minutes, causing the measurement to fail if ten or more errors are encountered. The entire analysis run is considered as Failed after three failed measurements.
+
+```yaml
+apiVersion: argoproj.io/v1alpha1
+kind: AnalysisTemplate
+metadata:
+  name: failure-condition-example
+spec:
+  metrics:
+  - name: total-errors
+    interval: 5m
+    failureCondition: result[0] >= 10
+    failureLimit: 3
+    provider:
+      prometheus:
+        address: http://prometheus.example.com:9090
+        query: |
+          sum(irate(
+            istio_requests_total{reporter="source",response_code=~"5.*"}[5m]
+          ))
+```
+
+
+## Delaying Measurements
+
+In some scenarios, it may be necessary to delay the start of a metric measurement. For example, some time may need to pass after an update in order for new data to populate in the monitoring services. The `initialDelay` option can be used to delay the start of measurements. Each metric can be configured to have a different delay.
+
+```yaml
+apiVersion: argoproj.io/v1alpha1
+kind: AnalysisTemplate
+metadata:
+  name: initial-delay-example
+spec:
+  metrics:
+  - name: success-rate
+    # Duration before measurement collection. Default is no delay
+    initialDelay: 5m
+    successCondition: result[0] >= 0.90
+    provider:
+      prometheus:
+        address: http://prometheus.example.com:9090
+        query: ...
+```
+
+## Example Metric Types
+
+### Web
+
+An HTTP request can be performed against some external service to obtain the measurements.
+
+```yaml
+apiVersion: argoproj.io/v1alpha1
+kind: AnalysisTemplate
+metadata:
+  name: web-metric-example
+spec:
+  args:
+  - name: api-token
+    valueFrom:
+      secretKeyRef:
+        name: token-secret
+        key: apiToken
+  metrics:
+  - name: webmetric
+    successCondition: result == true
+    provider:
+      web:
+        url: "http://example.com/api/v1/measurement"
+        # HTTP Method. valid values are GET|POST|PUT. Defaults to GET
+        method: POST
+        # Timeout for the request. Defaults to 10 seconds
+        timeoutSeconds: 20 
+        headers:
+        - key: Authorization
+          value: "Bearer {{ args.api-token }}"
+          # if body is a json, it is recommended to set the Content-Type
+        - key: Content-Type 
+          value: "application/json"
+        # Requst body to send. 
+        body: |
+          {"foo": "bar"}
+        # Optional JSON path to set the value of `result` in successCondition/failureCondition
+        jsonPath: "{$.data.ok}"
+```
+
+### Job
+
+A Kubernetes Job can be used to perform analysis. When a Job is used, the metric is considered successful if the Job completes and had an exit code of zero, otherwise it is failed.
```

Comment:
```suggestion
A Kubernetes `Job` can be used to perform analysis. When a `Job` is used, the metric is considered successful if the `Job` completes with an exit code of zero and is otherwise considered to have failed.
```